sdk/identity/Azure.Identity/src/OnBehalfOfCredential.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Microsoft.Identity.Client;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class OnBehalfOfCredential : TokenCredential
```

Comment:
AzureOnBehalfOfCredential?

Reply:
Good question - The credentials with that prefix all seem to be related to the name of the underlying component. Ex
- AzureCli => AzureClieCredential
- AzurePowerShell => AzurePowerShellCredential